packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -93,26 +97,53 @@ export class DefaultCrudRepository<T extends Entity, ID>
   ) {
     const definition = entityClass.definition;
     assert(
-      !!definition,
-      `Entity ${entityClass.name} must have valid model definition.`,
```

Comment:
Please preserve this assert and add your new assertion as a new `assert` call.

When `definition` is undefined, your check `definition.idProperties().length > 0` will crash on trying to access `idProperties` on an undefined object.